### PR TITLE
Reload dbus configuration after installing our service

### LINKED ISSUE #31: Install failure on Fedora34
I installed the alpha publish package from pip:

```
 jramsay  ~  $  pip install socranop                                                                                                                      
Defaulting to user installation because normal site-packages is not writeable                                                                                
Collecting socranop                                                                                                                                          
  Downloading socranop-0.4.92a1-py3-none-any.whl (71 kB)                                                                                                     
     |████████████████████████████████| 71 kB 5.0 MB/s                                                                                                       
Collecting pyusb                                                                                                                                             
  Downloading pyusb-1.2.1-py3-none-any.whl (58 kB)                                                                                                           
     |████████████████████████████████| 58 kB 9.7 MB/s                                                                                                       
Collecting pydbus                                                                                                                                            
  Downloading pydbus-0.6.0-py2.py3-none-any.whl (19 kB)                                                                                                      
Installing collected packages: pyusb, pydbus, socranop                                                                                                       
Successfully installed pydbus-0.6.0 pyusb-1.2.1 socranop-0.4.92a1
```

Then I ran the post-instal script:
```
 jramsay  ~  $  socranop-installtool --post-install
Using dirs HomeDirs(chroot=None, prefix='/home/jramsay/.local', datadir='/home/jramsay/.local/share', statedir='/home/jramsay/.config/socranop/state')
walk_through_data_files <socranop.installtool.BashCompletionInstallTool object at 0x7f61bf13a250> bash-completion                                            
res fullname data/bash-completion/socranop-session-service                                                                                                   
Adding file <socranop.installtool.BashCompletionInstallTool object at 0x7f61bf13a250> ResourceFile:/home/jramsay/.local/share/bash-completion/completions/soc
ranop-session-service:resource(data/bash-completion/socranop-session-service)                                                                                
res fullname data/bash-completion/socranop-gui                                                                                                               
Adding file <socranop.installtool.BashCompletionInstallTool object at 0x7f61bf13a250> ResourceFile:/home/jramsay/.local/share/bash-completion/completions/soc
ranop-gui:resource(data/bash-completion/socranop-gui)                                                                                                        
res fullname data/bash-completion/socranop-ctl                                                                                                               
Adding file <socranop.installtool.BashCompletionInstallTool object at 0x7f61bf13a250> ResourceFile:/home/jramsay/.local/share/bash-completion/completions/soc
ranop-ctl:resource(data/bash-completion/socranop-ctl)                                                                                                        
res fullname data/bash-completion/socranop-installtool                        
Adding file <socranop.installtool.BashCompletionInstallTool object at 0x7f61bf13a250> ResourceFile:/home/jramsay/.local/share/bash-completion/completions/soc
ranop-installtool:resource(data/bash-completion/socranop-installtool)                                                                                        
DBusInstallTool walk_through_data_files                                                                                                                      
walk_through_data_files <socranop.installtool.DBusInstallTool object at 0x7f61bf13a460> dbus-1                                                               
res fullname data/dbus-1/socranop.service                                                                                                                    
add_resource <socranop.installtool.DBusInstallTool object at 0x7f61bf13a460> data/dbus-1/socranop.service                                                    
Adding file <socranop.installtool.DBusInstallTool object at 0x7f61bf13a460> TemplateFile:/home/jramsay/.local/share/dbus-1/services/io.github.socratools.socr
anop.service:resource(data/dbus-1/socranop.service)
walk_through_data_files <socranop.installtool.ManpageInstallTool object at 0x7f61bfb70c10> man                                                               
res fullname data/man/socranop-session-service.1                              
Adding file <socranop.installtool.ManpageInstallTool object at 0x7f61bfb70c10> TemplateFile:/home/jramsay/.local/share/man/man1/socranop-session-service.1:re
source(data/man/socranop-session-service.1)                                                                                                                  
res fullname data/man/socranop-gui.1                                                                                                                         
Adding file <socranop.installtool.ManpageInstallTool object at 0x7f61bfb70c10> TemplateFile:/home/jramsay/.local/share/man/man1/socranop-gui.1:resource(data/
man/socranop-gui.1)                                                                                                                                          
res fullname data/man/socranop-ctl.1                                          
Adding file <socranop.installtool.ManpageInstallTool object at 0x7f61bfb70c10> TemplateFile:/home/jramsay/.local/share/man/man1/socranop-ctl.1:resource(data/
man/socranop-ctl.1)                                                           
walk_through_data_files <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> xdg                                                           
res fullname data/xdg/io.github.socratools.socranop.desktop                                                                                                  
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> TemplateFile:/home/jramsay/.local/share/applications/io.github.socratools.s
ocranop.desktop:resource(data/xdg/io.github.socratools.socranop.desktop)                                                                                     
res fullname data/xdg/io.github.socratools.socranop.24.png                                                                                                   
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/24x24/apps/io.github.
socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.24.png)                                                                              
res fullname data/xdg/io.github.socratools.socranop.64.png
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/64x64/apps/io.github.
socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.64.png)
res fullname data/xdg/io.github.socratools.socranop.48.png
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/48x48/apps/io.github.
socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.48.png)
res fullname data/xdg/io.github.socratools.socranop.192.png
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/192x192/apps/io.githu
b.socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.192.png)
res fullname data/xdg/io.github.socratools.socranop.svg
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/scalable/apps/io.gith
ub.socratools.socranop.svg:resource(data/xdg/io.github.socratools.socranop.svg)
res fullname data/xdg/io.github.socratools.socranop.128.png
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/128x128/apps/io.githu
b.socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.128.png)
res fullname data/xdg/io.github.socratools.socranop.256.png
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/256x256/apps/io.githu
b.socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.256.png)
res fullname data/xdg/io.github.socratools.socranop.16.png
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/16x16/apps/io.github.
socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.16.png)
res fullname data/xdg/io.github.socratools.socranop.96.png
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/96x96/apps/io.github.
socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.96.png)
res fullname data/xdg/io.github.socratools.socranop.32.png
Adding file <socranop.installtool.XDGDesktopInstallTool object at 0x7f61bf152310> ResourceFile:/home/jramsay/.local/share/icons/hicolor/32x32/apps/io.github.
socratools.socranop.png:resource(data/xdg/io.github.socratools.socranop.32.png)
Adding file <socranop.installtool.UdevRulesInstallTool object at 0x7f61bf152400> StringToFile:/etc/udev/rules.d/70-socranop.rules
Trying to import gi.repository.GLib
Trying to require module Gtk version 3.0
Trying to import gi.repository.Gtk
Trying to import gi.repository.Gio
Trying to require module GUdev version 1.0
Trying to import gi.repository.GUdev
Checking installed packages: Good.
  [inst] /home/jramsay/.local/share/bash-completion/completions/socranop-ctl
  [inst] /home/jramsay/.local/share/bash-completion/completions/socranop-gui
  [inst] /home/jramsay/.local/share/bash-completion/completions/socranop-installtool
  [inst] /home/jramsay/.local/share/bash-completion/completions/socranop-session-service
Old D-Bus service not running
  [inst] /home/jramsay/.local/share/dbus-1/services/io.github.socratools.socranop.service
Starting D-Bus service io.github.socratools.socranop as a test...
Installtool version: 0.4.92
Retrying 5...
Retrying 4...
Retrying 3...
Retrying 2...
Retrying 1...
Traceback (most recent call last):
  File "/home/jramsay/.local/bin/socranop-installtool", line 8, in <module>
    sys.exit(main())
  File "/home/jramsay/.local/lib/python3.9/site-packages/socranop/installtool.py", line 890, in main
    everything.post_install()
  File "/home/jramsay/.local/lib/python3.9/site-packages/socranop/installtool.py", line 813, in post_install
    thing.post_install()
  File "/home/jramsay/.local/lib/python3.9/site-packages/socranop/installtool.py", line 554, in post_install
    dbus_service.StartServiceByName(const.BUSNAME, 0)
  File "/home/jramsay/.local/lib/python3.9/site-packages/pydbus/proxy_method.py", line 72, in __call__
    ret = instance._bus.con.call_sync(
gi.repository.GLib.Error: g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable (2)
```

Not only did the dbus service not start, but I also don't see the .desktop file or the icons where they're supposed to be:

```
 jramsay  ~  $  ls .local/share/applications/
chrome-aapocclcgogkmnckokdopfmhonfmgoek-Profile_1.desktop  chrome-mdpkiolbdkhdjpekfbkbmhigcaggjagi-Profile_1.desktop
chrome-aghbiahbpaijignceidepookljebhfak-Default.desktop    chrome-nmmhkkegccagdldgiimedpiccmgmieda-Default.desktop
chrome-agimnkijcaahngcdmfeangaknmldooml-Profile_1.desktop  chrome-nmmhkkegccagdldgiimedpiccmgmieda-Profile_1.desktop
chrome-aohghmighlieiainnegkcijnfilokake-Profile_1.desktop  chrome-nmmhkkegccagdldgiimedpiccmgmieda-Profile_2.desktop
chrome-apdfllckaahabafndbhieahigkjlhalf-Default.desktop    chrome-nmmhkkegccagdldgiimedpiccmgmieda-Profile_3.desktop
chrome-apdfllckaahabafndbhieahigkjlhalf-Profile_1.desktop  chrome-pjkljhegncpnkpknbcohdijeoejaedia-Default.desktop
chrome-cinhimbnkkaeohfgghhklpknlkffjgod-Profile_1.desktop  chrome-pjkljhegncpnkpknbcohdijeoejaedia-Profile_1.desktop
chrome-fahmaaghhglfmonjliepjlchgpgfmobi-Profile_4.desktop  install4j_j162e7-FTBApp.desktop
chrome-felcaaldnbdncclmgdcncolpebgiejap-Profile_1.desktop  mimeapps.list
chrome-hmjkmjkepdijhoojdojkdfohbdgmmhki-Profile_1.desktop  mimeinfo.cache
chrome-hpfldicfbfomlpcikngkocigghgafkph-Profile_1.desktop  Webcamoid.desktop
chrome-mdpkiolbdkhdjpekfbkbmhigcaggjagi-Default.desktop
 jramsay  ~  $  ls .local/share/icons/hicolor/128x128/apps/
chrome-aapocclcgogkmnckokdopfmhonfmgoek-Profile_1.png  chrome-fhbjgbiflinjbdggehcddcbncdddomop-Profile_1.png
chrome-aghbiahbpaijignceidepookljebhfak-Default.png    chrome-hmjkmjkepdijhoojdojkdfohbdgmmhki-Profile_1.png
chrome-agimnkijcaahngcdmfeangaknmldooml-Profile_1.png  chrome-hpfldicfbfomlpcikngkocigghgafkph-Profile_1.png
chrome-aohghmighlieiainnegkcijnfilokake-Profile_1.png  chrome-mdpkiolbdkhdjpekfbkbmhigcaggjagi-Default.png
chrome-apdfllckaahabafndbhieahigkjlhalf-Default.png    chrome-mdpkiolbdkhdjpekfbkbmhigcaggjagi-Profile_1.png
chrome-apdfllckaahabafndbhieahigkjlhalf-Profile_1.png  chrome-nmmhkkegccagdldgiimedpiccmgmieda-Default.png
chrome-chfbpgnooceecdoohagngmjnndbbaeip-Default.png    chrome-nmmhkkegccagdldgiimedpiccmgmieda-Profile_1.png
chrome-chfbpgnooceecdoohagngmjnndbbaeip-Profile_1.png  chrome-nmmhkkegccagdldgiimedpiccmgmieda-Profile_2.png
chrome-cinhimbnkkaeohfgghhklpknlkffjgod-Profile_1.png  chrome-nmmhkkegccagdldgiimedpiccmgmieda-Profile_3.png
chrome-fahmaaghhglfmonjliepjlchgpgfmobi-Profile_1.png  chrome-pjkljhegncpnkpknbcohdijeoejaedia-Default.png
chrome-fahmaaghhglfmonjliepjlchgpgfmobi-Profile_4.png  chrome-pjkljhegncpnkpknbcohdijeoejaedia-Profile_1.png
chrome-felcaaldnbdncclmgdcncolpebgiejap-Profile_1.png  webcamoid.png
```

### PR DESCRIPTION
**Pull request by lack**
Fixes #31

### CODE REVIEW

**Review by ndim — COMMENTED** [review]
Doesn't this introduce a race between asking the bus for the reload and asking for the list of bus activatable services? A race which this leaves not much time for to be won? Does this need a similar countdown race workaround like starting the service has one?

**Review by ndim — CHANGES_REQUESTED** [review]
Shouldn't reloading the DBus fall under the not(no_launch) condition?

**Comment by lack** [comment]
> Doesn't this introduce a race between asking the bus for the reload and asking for the list of bus activatable services? A race which this leaves not much time for to be won? Does this need a similar countdown race workaround like starting the service has one?

There is potentially a race, depending on how the reload command is implemented, but because the list check is non-fatal, it's more of a warning to help diagnose later failure.  The retry loop below is good enough.

I suppose we could have:
- a single reload
- a timed loop around checking for the name to appear
- a single activation attempt

**Comment by lack** [comment]
> Shouldn't reloading the DBus fall under the not(no_launch) condition?

Good question.  If the point of the flag is just that we don't want to launch the service, no.  If the point is that we don't want any interaction with the runtime system (for example, because we're working on a chroot), yes (but maybe we should rename the flag if that's the case?)

**Comment by lack** [comment]
I think this last push takes care of both your concerns.

**Comment by ndim** [comment]
> > Shouldn't reloading the DBus fall under the not(no_launch) condition?
> 
> Good question. If the point of the flag is just that we don't want to launch the service, no. If the point is that we don't want any interaction with the runtime system (for example, because we're working on a chroot), yes (but maybe we should rename the flag if that's the case?)

There are both `--no-launch` and `--chroot CHROOT` arguments. Historically, `--no-launch` came first and was for avoiding interaction with the runtime system, to help with the use cases which are just preparing files on disk. Back then, the only interaction with the runtime system was to launch the D-Bus service. Then came `--chroot CHROOT` to allow preparing a full installation in a chroot for e.g. building RPMs, which implies `--no-launch`. And then some time, shutting down old services etc. was added.

There might also be packaging systems which want to avoid the interaction with the runtime system, but do not use a chroot environment, so for these, the `--no-launch` functionality might still be useful. There might be better names. `--filesystem-only` perhaps?

**Comment by ndim** [comment]
During the day, I had the following idea for the D-Bus service post-install:

    shutdown_running_service()
    install_service_config_file()
    tell_dbus_to_reload()
    while True:
        try:
            our_service = bus.get(...)
            if our_service.version != installtool_version:
                raise VersionMismatch()
            if our_service.socranop_path != socranop.__path__:
                raise InstallLocationMismatch()
            break
        catch gi.repository.GLib.Error:
            timeout_counter -= 1
            if timeout_counter == 0:
                raise DBusServiceCannotBeActivated()
            sleep(1_s)
    shutdown_running_service() # let installer leave no processes running

While the module path definitively should not be part of the D-Bus interface for the general public, comparing the location the service process loads its `socranop` module from and the location the installtool loads its `socranop` module from should help catch things where the version number is nominally the same, but there is e.g. a git clone being installed into $HOME, while there already is a system package in /usr.

From what I have just seen, your proposed timeout loop makes more sense in many aspects than what I pseudocoded above, but there are still aspects of this which appear useful.

**Review by ndim — APPROVED** [review]

File: socranop/installtool.py [hunk 1]
```diff
@@ -523,63 +523,74 @@ def add_resource(self, fullname):
...
+        dbus_service.ReloadConfig()
+
+        (attempt, max_attempt) = (0, 5)
+        while const.BUSNAME not in dbus_service.ListActivatableNames():
+            attempt += 1
+            if attempt > max_attempt:
+                raise RuntimeError(
+                    f"The dbus service we just installed is not yet detected after {max_attempt}s. You may need to restart your dbus session (for example, by logging out and back in to your desktop)"
+                )
+            print(f"Waiting for our service to appear ({attempt}/{max_attempt})...")
```
Comment (ndim) [inline socranop/installtool.py:578]: Would this be better?

```python
print(f"Waiting for our service to become activatable ({attempt}/{max_attempt})...")
```

**Pull request merged**